File: course/section_5_binary_search/code/03-search-insert-position.py
from typing import List


class Solution:
    def searchInsert(self, nums: List[int], target: int) -> int:
        if nums[0] > target:
            return 0
        if nums[-1] < target:
            return len(nums)
        l, r = 0, len(nums) - 1
        m = 0
        while l <= r:
            m = (l + r) // 2
            if nums[m] > target:
                r = m - 1
            elif nums[m] < target:
                l = m + 1
            else:
                return m
        return l
        # try:
        #     return m if nums[m] > target and nums[m + 1] < target else m + 1
        # except:
        #     return m

File: course/section_5_binary_search/code/03-search-insert-position.php
<?php

class Solution {

    function searchInsert($nums, $target) {
        $left = 0 ;
        $right = count($nums) - 1  ;
        while($left<=$right){
            $m = intdiv($left + $right , 2) ;
            $mid = $nums[$m] ;
            if($mid === $target){
                return $m ;
            } elseif($mid > $target){
                $right = $m - 1 ;
            } else {
                $left = $m + 1 ;
            }
        }
        return $left ;
    }
}